README.md
## bash  
  
Bash is the GNU Project's shell  
  
requires:

```shell
apt install bash bash-completion direnv
```  

```shell
yum install bash bash-completion direnv
```  

```shell
pacman -S bash bash-completion direnv
```  

```shell
xbps-install -S bash bash-completion direnv
```

```shell
brew install bash bash-completion direnv
```

Automatic install/update:

```shell
bash -c "$(curl -LSs https://github.com/dfmgr/bash/raw/main/install.sh)"
```

Manual install:

```shell
mv -fv "$HOME/.config/bash" "$HOME/.config/bash.bak"
git clone https://github.com/dfmgr/bash "$HOME/.config/bash"
ln -sf $HOME/.config/bash/bashrc $HOME/.bashrc
ln -sf $HOME/.config/bash/bash_logout $HOME/.bash_logout
ln -sf $HOME/.config/bash/bash_profile $HOME/.bash_profile

```
  
  
<p align=center>
  <a href="https://wiki.archlinux.org/index.php/bash" target="_blank">bash wiki</a>  |  
  <a href="https://www.gnu.org/software/bash/" target="_blank">bash site</a>
</p>  


CHANGELOG.md
#### Changelog: 2023-04-13 at 13:48:11  

etc/plugins/asdf.bash : 🗃️  Committing everything that changed 🗃️ 🚀  
etc/plugins/basher.bash : 🗃️  Committing everything that changed 🗃️ 🚀  
etc/plugins/luaver.bash : 🗃️  Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202304111220-git - 2023-04-13 at 13:48:11  
  
----  
  
#### Changelog: 2023-04-11 at 12:20:55  

etc/functions/python.bash : 🗃️  Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202304111151-git - 2023-04-11 at 12:20:55  
  
----  
  
#### Changelog: 2023-04-11 at 11:51:19  

etc/functions/python.bash : 🗃️  Committing everything that changed 🗃️ 🚀  
etc/prompt/01-powerline.bash : 🗃️  Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202304031301-git - 2023-04-11 at 11:51:19  
  
----  
  
#### Changelog: 2023-04-03 at 13:01:54  

etc/functions/command-not-found.bash : 🗃️  Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202303131418-git - 2023-04-03 at 13:01:54  
  
----  
  
#### Changelog: 2023-03-13 at 14:18:04  

completions/add2path.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
etc/functions/add2path.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202303081037-git - 2023-03-13 at 14:18:04  
  
----  
  
#### Changelog: 2023-03-08 at 10:37:07  

etc/bashrc : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202303031037-git - 2023-03-08 at 10:37:07  
  
----  
  
#### Changelog: 2023-03-03 at 10:37:42  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202303031034-git - 2023-03-03 at 10:37:42  
  
----  
  
#### Changelog: 2023-03-03 at 10:34:59  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202303022204-git - 2023-03-03 at 10:34:59  
  
----  
  
#### Changelog: 2023-03-02 at 22:04:58  

etc/aliases/00-default.lin : 🗃️ Committing everything that changed 🗃️ 🚀  
etc/aliases/00-default.mac : 🗃️ Committing everything that changed 🗃️ 🚀  
etc/aliases/00-default.win : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202302251748-git - 2023-03-02 at 22:04:58  
  
----  
  
#### Changelog: 2023-02-25 at 17:48:35  

etc/aliases/00-default.lin : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202210171955-git - 2023-02-25 at 17:48:35  
  
----  
  
#### Changelog: 2022-10-17 at 19:55:07  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202210171926-git - 2022-10-17 at 19:55:07  
  
----  
  
#### Changelog: 2022-10-17 at 19:26:21  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202210171713-git - 2022-10-17 at 19:26:21  
  
----  
  
#### Changelog: 2022-10-17 at 17:13:37  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202210171557-git - 2022-10-17 at 17:13:37  
  
----  
  
#### Changelog: 2022-10-17 at 15:57:34  

etc/functions/00-functions.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
etc/functions/add2path.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
etc/functions/dirignore.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
etc/functions/systeminfo.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
etc/functions/tempature.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202210171412-git - 2022-10-17 at 15:57:34  
  
----  
  
#### Changelog: 2022-10-17 at 14:12:47  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202210152338-git - 2022-10-17 at 14:12:47  
  
----  
  
#### Changelog: 2022-10-15 at 23:38:49  

applications/bash.desktop : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202209200934-git - 2022-10-15 at 23:38:49  
  
----  
  
#### Changelog: 2022-09-20 at 09:34:53  

applications/bash.desktop : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202209200928-git - 2022-09-20 at 09:34:53  
  
----  
  
#### Changelog: 2022-09-20 at 09:28:46  

applications/bash.desktop : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202209091508-git - 2022-09-20 at 09:28:46  
  
----  
  
#### Changelog: 2022-09-09 at 15:08:11  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202208281202-git - 2022-09-09 at 15:08:11  
  
----  
  
#### Changelog: 2022-08-28 at 12:02:44  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202208281149-git - 2022-08-28 at 12:02:44  
  
----  
  
#### Changelog: 2022-08-28 at 11:49:00  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202208281124-git - 2022-08-28 at 11:49:00  
  
----  
  
#### Changelog: 2022-08-28 at 11:24:28  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202208281039-git - 2022-08-28 at 11:24:28  
  
----  
  
#### Changelog: 2022-08-28 at 10:39:27  

etc/prompt/01-powerline.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202208271944-git - 2022-08-28 at 10:39:27  
  
----  
  
#### Changelog: 2022-08-27 at 19:44:18  

etc/functions/python.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202208271848-git - 2022-08-27 at 19:44:18  
  
----  
  
#### Changelog: 2022-08-27 at 18:48:59  

etc/functions/python.bash : 🗃️ Committing everything that changed 🗃️ 🚀  
  
#### End of changes for 202208271510-git - 2022-08-27 at 18:48:59  
  
----  
  
#### Changelog: 2022-08-27 at 15:10:42  

etc/aliases/00-default.lin : 🗃️ Removed unused aliases 🗃️ 🚀  

install.sh
#!/usr/bin/env bash
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
APPNAME="bash"
USER="${SUDO_USER:-${USER}}"
HOME="${USER_HOME:-${HOME}}"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
#set opts

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
##@Version       : 020820212259-git
# @Author        : Jason Hempstead
# @Contact       : jason@casjaysdev.com
# @License       : LICENSE.md
# @ReadME        : README.md
# @Copyright     : Copyright: (c) 2021 Jason Hempstead, CasjaysDev
# @Created       : Monday, Feb 08, 2021 22:59 EST
# @File          : bash
# @Description   : Installer script for bash
# @TODO          :
# @Other         :
# @Resource      :
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Import functions
CASJAYSDEVDIR="${CASJAYSDEVDIR:-/usr/local/share/CasjaysDev/scripts}"
SCRIPTSFUNCTDIR="${CASJAYSDEVDIR:-/usr/local/share/CasjaysDev/scripts}/functions"
SCRIPTSFUNCTFILE="${SCRIPTSAPPFUNCTFILE:-app-installer.bash}"
SCRIPTSFUNCTURL="${SCRIPTSAPPFUNCTURL:-https://github.com/dfmgr/installer/raw/main/functions}"
connect_test() { ping -c1 1.1.1.1 &>/dev/null || curl --disable -LSs --connect-timeout 3 --retry 0 --max-time 1 1.1.1.1 2>/dev/null | grep -e "HTTP/[0123456789]" | grep -q "200" -n1 &>/dev/null; }
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
if [ -f "$PWD/$SCRIPTSFUNCTFILE" ]; then
  . "$PWD/$SCRIPTSFUNCTFILE"
elif [ -f "$SCRIPTSFUNCTDIR/$SCRIPTSFUNCTFILE" ]; then
  . "$SCRIPTSFUNCTDIR/$SCRIPTSFUNCTFILE"
elif connect_test; then
  curl -LSs "$SCRIPTSFUNCTURL/$SCRIPTSFUNCTFILE" -o "/tmp/$SCRIPTSFUNCTFILE" || exit 1
  . "/tmp/$SCRIPTSFUNCTFILE"
else
  echo "Can not load the functions file: $SCRIPTSFUNCTDIR/$SCRIPTSFUNCTFILE" 1>&2
  exit 1
fi
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Call the main function
user_installdirs
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# OS Support: supported_os unsupported_oses
unsupported_oses
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Make sure the scripts repo is installed
scripts_check
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Defaults
APPNAME="${APPNAME:-bash}"
APPDIR="$CONF/$APPNAME"
INSTDIR="$CASJAYSDEVSHARE/$SCRIPTS_PREFIX/$APPNAME"
REPO_BRANCH="${GIT_REPO_BRANCH:-main}"
REPO="$DFMGRREPO/$APPNAME"
REPORAW="$REPO/raw/$REPO_BRANCH"
APPVERSION="$(__appversion "$REPORAW/version.txt")"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Setup plugins
PLUGNAMES=""
PLUGDIR="${SHARE:-$HOME/.local/share}/$APPNAME"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Call the dfmgr function
dfmgr_install
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Script options IE: --help --version
show_optvars "$@"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Do not update
#installer_noupdate "$@"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Requires root - no point in continuing
#sudoreq  # sudo required
#sudorun  # sudo optional
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# initialize the installer
dfmgr_run_init
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# end with a space
APP="$APPNAME "
PERL=""
PYTH=""
PIPS=""
CPAN=""
GEMS=""
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# install packages - useful for package that have the same name on all oses
install_packages "$APP"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# install required packages using file
install_required "$APP"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# check for perl modules and install using system package manager
install_perl "$PERL"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# check for python modules and install using system package manager
install_python "$PYTH"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# check for pip binaries and install using python package manager
install_pip "$PIPS"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# check for cpan binaries and install using perl package manager
install_cpan "$CPAN"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# check for ruby binaries and install using ruby package manager
install_gem "$GEMS"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Other dependencies
dotfilesreq git misc
dotfilesreqadmin
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Ensure directories exist
ensure_dirs
ensure_perms
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Backup if needed
if [ -d "$APPDIR" ]; then
  execute "backupapp $APPDIR $APPNAME" "Backing up $APPDIR"
fi
# Main progam
if am_i_online; then
  if [ -d "$INSTDIR/.git" ]; then
    execute "git_update $INSTDIR" "Updating $APPNAME configurations"
  else
    execute "git_clone $REPO $INSTDIR" "Installing $APPNAME configurations"
  fi
  # exit on fail
  failexitcode $? "Failed to download $REPO/$APPNAME to $INSTDIR"
fi
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Plugins
if am_i_online; then
  if [ "$PLUGNAMES" != "" ]; then
    if [ -d "$PLUGDIR/" ]; then
      execute "git_update $PLUGDIR/" "Updating plugin "
    else
      execute "git_clone REPO $PLUGDIR/" "Installing plugin "
    fi
  fi
  # exit on fail
  failexitcode $? "Failed to download Plugin repo"
fi
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# run post install scripts
run_postinst() {
  history -r &>/dev/null
  history -a &>/dev/null
  history -w &>/dev/null
  history -a &>/dev/null
  if [ -f "$HOME/.config/bash/bash_history" ]; then mv_f "$HOME/.config/bash/bash_history" "/tmp/bash_history.tmp"; fi
  if [ -f "$APPDIR/welcome.msg" ]; then WELCOME=true; fi
  for file in aliases bash_logout bash_profile bashrc completions exports plugins profile prompt; do
    rm_rf "$APPDIR/$file"
  done
  dfmgr_run_post
  ln_sf "$APPDIR/bashrc" "$HOME/.bashrc"
  ln_sf "$APPDIR/bash_logout" "$HOME/.bash_logout"
  ln_sf "$APPDIR/bash_profile" "$HOME/.bash_profile"
  [ ! -f "$COMPDIR/README.md" ] || rm_rf "$COMPDIR/README.md"
  if [ -n "$WELCOME" ]; then touch "$APPDIR/welcome.msg"; fi
  if [ -f "$HOME/.bash_history" ] && [ ! -e "HOME/.config/bash/bash_history" ]; then
    mv -f "$HOME/.bash_history" "$HOME/.config/bash/bash_history"
  elif [ -f "$HOME/.bash_history" ] && [ -e "HOME/.config/bash/bash_history" ]; then
    cat "$HOME/.bash_history" >>"$HOME/.config/bash/bash_history"
    rm -Rf "$HOME/.bash_history"
  elif [[ -f "/tmp/bash_history.tmp" ]]; then
    mv -f "/tmp/bash_history.tmp" "$HOME/.config/bash/bash_history"
  fi
  [[ -f "$HOME/.bashrc" ]] && . "$HOME/.bashrc"
  history -a &>/dev/null
  history -w &>/dev/null
  history -a &>/dev/null
  history -r &>/dev/null
}
#
execute "run_postinst" "Running post install scripts"
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# create version file
dfmgr_install_version
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# exit
run_exit
# end
